src/Http/Routing/src/RouteEndpointBuilder.cs

```diff
@@ -95,6 +103,22 @@ private static EndpointMetadataCollection CreateMetadataCollection(IList<object>
             }
         }
 
+        // No route diagnostics metadata provided to automatically add one based on the route pattern string.
+        if (!hasRouteDiagnosticsMetadata)
+        {
+            metadata.Add(new RouteDiagnosticsMetadata(routePattern.DebuggerToString()));
```

Comment:
`DebuggerToString` doesn't sound like something that should be published?  Would this prevent us from changing the format of `DebuggerToString` in the future?

Reply:
We need a string representation of the route, and `DebuggerToString` provides that. I don't see that ever changing in the future. If it does, there are metrics unit tests that assert the route tag.

Reply:
Added a comment to DebuggerToString. I discovered it's already used in the route display name.